sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/CredentialBuilderBase.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+import com.azure.core.http.ProxyOptions;
+import com.azure.identity.implementation.IdentityClientOptions;
+
+import java.util.function.Function;
+
+/**
+ * The base class for all the credential builders.
+ * @param <T> the type of the credential builder
+ */
+public abstract class CredentialBuilderBase<T extends CredentialBuilderBase<T>> {
+    IdentityClientOptions identityClientOptions;
+
+    CredentialBuilderBase() {
+        this.identityClientOptions = new IdentityClientOptions();
+    }
+
+    /**
+     * Specifies the Azure Active Directory endpoint to acquire tokens.
+     * @param authorityHost the Azure Active Directory endpoint
+     * @return {@link <T>} itself
+     */
+    @SuppressWarnings("unchecked")
+    public T authorityHost(String authorityHost) {
+        this.identityClientOptions.authorityHost(authorityHost);
+        return (T) this;
+    }
```

Comment:
authority host doesn't make sense for some of the credential builders which derive from this such as ManagedIdentityCredentialBuilder

Reply:
Moved to CredentialBuilderBase